app/assets/javascripts/handin.js

```diff
@@ -50,6 +50,24 @@ function showFiles() {
   var fileSelector = $("#handin_show_assessment input[type='file']").get(0);
   var file = fileSelector.files[0];
   $("#handin-file-name").text(file.name);
+
+  // only do check for file type that has a period
+  if( $('#handin-file-type').length && file
+     && file.name.split(".").length >= 2)         // use this if you are using id to check
+  {
+    $('#handin-file-type-incorrect').text("")
+    var file_type= file.name.split(".")[file.name.split(".").length -1];
```

Comment:
Nit: space before `=` and after `-`

We should probably concatenate from the first position onwards, otherwise extensions such as `.tar.gz` wouldn't work. However this means that we can't handle file with extraneous periods

Reply:
done